src/full/Agda/TypeChecking/Rules/Application.hs

```diff
@@ -964,9 +964,9 @@ checkProjApp cmp e o ds args0 t = do
 -- | Checking the type of an overloaded projection application.
 --   See 'inferOrCheckProjAppToKnownPrincipalArg'.
 
-checkProjAppToKnownPrincipalArg  :: Comparison -> A.Expr -> ProjOrigin -> List1 QName -> A.Args -> Type -> Int -> Term -> Type -> TCM Term
-checkProjAppToKnownPrincipalArg cmp e o ds args0 t k v0 pt = do
-  (v, ti, targetCheck) <- inferOrCheckProjAppToKnownPrincipalArg e o ds args0 (Just (cmp, t)) k v0 pt
+checkProjAppToKnownPrincipalArg  :: Comparison -> A.Expr -> ProjOrigin -> List1 QName -> A.Args -> Type -> Int -> Term -> Type -> (Args, Type) -> TCM Term
```

Comment:
Please document the purpose of the two new arguments.

Reply:
I am a little reluctant to do that, as the existing comment on this function basically says "for more information, take a look at `inferOrCheckProjAppToKnownPrincipalArg`". This makes it consistent with the surrounding functions `inferProjApp` and `checkProjApp`, which do the same thing and which seem to be on the same level of abstraction as `checkProjAppToKnownPrincipalArg`. So, maybe opt for consistency here? Happy to add the comment, though, if you'd prefer that.

Reply:
Fine with me.